question19/js/slideshow.js

```diff
@@ -0,0 +1,184 @@
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = "loading";
+  imgListsWrapper.appendChild(loadImg);
+};
+
+const removeLoadImg = () => {
+  document.getElementById("loading").remove();
+};
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch("https://mocki.io/v1/09593e78-80ba-4765-940e-3c70fb46078c"));
+  }, 3000);
+});
+
+//Get json data
+const callApi = async () => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  } catch (error) {
+    console.error(error);
+  } finally {
+    removeLoadImg();
+  }
+};
+
+const init = async () => {
+  const imgData = await callApi();
+  if (imgData) {
+    renderImgUiElement(imgData);
+  } else {
+    imgListsWrapper.textContent = "データの表示に失敗しました";
+    console.error("データの表示に失敗しました");
+  }
+};
+init();
+
+const renderImgUiElement = (imgData) => {
+  createBtnElements(imgData);
+  renderPageNumElement(imgData);
+  renderPagenation(imgData);
+  clickedPagenation(imgData);
+  autoImgSwitch(imgData);
+  imgLists.appendChild(getFragmentImglists(imgData));
+};
+
+const renderPageNumElement = (imgData) => {
+  const numberOfPage = createElementWithClassName("p", "pageNumber");
+  imgListsWrapper.appendChild(numberOfPage);
+  numberOfPage.id = "js-number";
+  numberOfPage.textContent = `1 / ${imgData.length}`;
+};
+
+const getFragmentImglists = (imgData) => {
+  const fragmentImglists = document.createDocumentFragment();
+  for (let i = 0; i < imgData.length; i++) {
+    const imgList = createElementWithClassName("li", "imgList");
+    const img = createElementWithClassName("img", `img_0${[i + 1]}`);
+    imgList.dataset.index = i;
+    img.src = imgData[i].img;
+    img.alt = imgData[i].description;
+    i === 0 && imgList.classList.add("is-show");
+    fragmentImglists.appendChild(imgList).appendChild(img);
+  }
+  return fragmentImglists;
+};
+
+const imgNum = { count: 0 };
+const imgList = document.getElementsByClassName("imgList");
+
+const createBtnElements = (imgData) => {
+  const btnDirections = ["next", "prev"];
+
+  btnDirections.forEach((btnDirections) => {
+    const btn = document.createElement("button");
+    btn.classList.add(`${btnDirections}`);
+    btn.id = btnDirections === "prev" ? "js-prevbtn" : "js-nextbtn";
+    btn.textContent = btnDirections === "prev" ? "◀︎" : "▶︎";
+    imgListsWrapper.appendChild(btn);
+    btn.disabled = btnDirections === "prev";
+
+    btn.addEventListener("click", function () {
+      btnDirections === "prev" ? (imgNum.count -= 1) : (imgNum.count += 1);
+      document.querySelector(".is-show").classList.remove("is-show");
+      imgList[imgNum.count].classList.add("is-show");
+      switchPagenation(imgNum.count);
+      switchDisableForBtn(imgData);
+      setNumberOfPage(imgData);
+      resetAutoPlay(imgData);
```

Comment:
Q
I felt that it was better to cut out this part. However, since the button is a pusher in the first place, was the part that creates the button and the part that creates the event combined into the same function?

Reply:
Thank you for your review!! I didn't have the idea when I created it at the time, but I certainly felt it would be better to separate them. I tried to separate createArrowBtnsElements() and arrowBtnsClickEvent(). I hope you can confirm this.
fixex commit : https://github.com/csakyo/homework/pull/33/commits/c57fdc7f532d364cf76a3304b25b0811b5f466da